### PR TITLE
Update Blob::{new, new_inherited} to take Strings

### LINKED ISSUE #11762: Blob::{new, new_inherited} should take String, not &str.
Right now these functions take `&str` and copy the string into the `typeString` field. It would be better to change the argument to be a `String`, so the callers can pass in an existing `String` if they have one. (`Blob::Constructor` does.)

* Code: `components/script/dom/blob.rs`.
* Test: if it compiles, that's enough

### PR DESCRIPTION
**Pull request by achals**
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #11762.

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because no logic changes, only interface changes.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11768)

### CODE REVIEW

File: components/script/dom/blob.rs [hunk 1]
```diff
@@ -156,7 +156,7 @@ impl Blob {
         };
 
         let slice = DataSlice::from_bytes(bytes);
-        Ok(Blob::new(global, BlobImpl::new_from_slice(slice), &blobPropertyBag.get_typestring()))
+        Ok(Blob::new(global, BlobImpl::new_from_slice(slice), blobPropertyBag.get_typestring().to_owned()))
```
Comment (Ms2ger) [inline components/script/dom/blob.rs]: You should be able to remove this to_owned call

File: components/script/dom/blob.rs [hunk 2]
```diff
@@ -252,7 +252,7 @@ impl BlobMethods for Blob {
         let global = self.global();
         let bytes = self.get_slice_or_empty().bytes.clone();
         let slice = DataSlice::new(bytes, start, end);
-        Blob::new(global.r(), BlobImpl::new_from_slice(slice), &relativeContentType)
+        Blob::new(global.r(), BlobImpl::new_from_slice(slice), relativeContentType.to_string())
```
Comment (Ms2ger) [inline components/script/dom/blob.rs]: It would be better to use `.into()` here

**Comment by Ms2ger** [comment]
Nice job, just two small comments.

**Comment by achals** [comment]
Neato, updating.

**Comment by Ms2ger** [comment]
Thank you!

@bors-servo r+

**Pull request merged**